### PR TITLE
feat: support for downloading original images

### LINKED ISSUE #6120: Quality of downloaded NFTs
I would love to ask about the quality of downloaded NFTs on KodaDot:
Screenshot 2023-05-28 at 10 49 09

I downloaded one NFT, and the file's size and quality were very low (177MB). I'm just curious if this is the quality artist must have uploaded, or do we somehow lower the quality of files on purpose?

### PR DESCRIPTION
**Pull request by newraina**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #6120 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [x] I've tested it at [/ksm/gallery/13902891-008963f0b21767d825-BULL-RMRK_BULL_144-00000144](https://deploy-preview-6958--koda-canary.netlify.app/ksm/gallery/13902891-008963f0b21767d825-BULL-RMRK_BULL_144-00000144), and got the original image (995KB)
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce&usdamount=0&donation=true)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 85e1079</samp>

Added a feature to download the original image from IPFS in the gallery item menu. Implemented a new function `toOriginalContentUrl` in `utils/ipfs.ts` and used it in `GalleryItemMoreActionBtn.vue`.

### <samp>🤖 Generated by Copilot at 85e1079</samp>

> _`toOriginalContentUrl`_
> _Adds a query string_
> _For IPFS downloads_

### CODE REVIEW

File: utils/ipfs.ts [hunk 1]
```diff
@@ -182,3 +182,10 @@ export function toCloudflareIpfsUrl(baseurl) {
   const url = new URL(baseurl)
   return `https://cloudflare-ipfs.com/${url.pathname}`
 }
+
+export function toOriginalContentUrl(baseurl: string) {
+  const hasQueryString = baseurl.includes('?')
+  const separator = hasQueryString ? '&' : '?'
```
Comment (preschian) [inline utils/ipfs.ts:188]: better use [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL_API)

```js
const url = new URL(baseurl)
url.searchParams.append('original', 'true')

return url.toString()

```

**Review by preschian — CHANGES_REQUESTED** [review]

**Review by preschian — APPROVED** [review]

**Comment by prury** [comment]
tested on rmrk | assethubs | basilisk
working ✔️

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.45 USD/DOT ~ 6.742 $DOT
🧗 [1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce](https://polkadot.subscan.io/account/1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce)
🔗 [0x231116d575a84bfc4268ce4b388c447b52dff9ac191101d602f8fc523929bbbd](https://polkadot.subscan.io/extrinsic/0x231116d575a84bfc4268ce4b388c447b52dff9ac191101d602f8fc523929bbbd)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**